### ISSUE TITLE
robotgo is incompatible with Go 1.21 

### ISSUE THREAD

**Issue by ChristoWolf** [go-vgo/robotgo#621]
- Robotgo version (or commit ref): `v0.100.10`
- Go version: `1.21` (and up)
- Gcc version: `13.2.0` included in WinLibs 11.0.1 (UCRT)
- Operating system and bit: Version 22H2 (OS Build 19045.3324), 64bit
- Resolution: not relevant
- Can you reproduce the bug at [Examples](https://github.com/go-vgo/robotgo/blob/master/examples/main.go):
  - Yes, as `zlib` and `zconf` is required by robotgo.

- Log gist (when compiling):
```
.\go\1.21.4\x64\bin\go.exe build -v
github.com/go-vgo/robotgo/clipboard
github.com/lxn/win
github.com/vcaesar/keycode
runtime/cgo
github.com/go-ole/go-ole
github.com/go-ole/go-ole/oleutil
github.com/StackExchange/wmi
github.com/shirou/gopsutil/internal/common
github.com/shirou/gopsutil/cpu
github.com/shirou/gopsutil/mem
github.com/shirou/gopsutil/net
github.com/shirou/gopsutil/process
github.com/vcaesar/gops
image/color
image
golang.org/x/image/bmp
hash/adler32
compress/zlib
golang.org/x/image/ccitt
golang.org/x/image/tiff/lzw
golang.org/x/image/tiff
image/color/palette
image/internal/imageutil
image/draw
image/gif
image/jpeg
image/png
github.com/vcaesar/imgo
github.com/vcaesar/tt
github.com/robotn/gohook
encoding/csv
github.com/spf13/pflag
github.com/go-vgo/robotgo
# github.com/go-vgo/robotgo
In file included from ./bitmap/../base/str_io_c.h:2,
                 from ./bitmap/goBitmap.h:17,
                 from C:\Users\<some user>\go\pkg\mod\github.com\go-vgo\robotgo@v0.100.10\bitmap.go:24:
./bitmap/../base/zlib_util_c.h:2:10: fatal error: zlib.h: No such file or directory
    2 | #include <zlib.h>
      |          ^~~~~~~~
compilation terminated.
```

## Description

Hi!

Go 1.21 and up requires newer GCC/MinGW toolchains [see this and related issues](https://github.com/golang/go/issues/61916), so installing [MinGW-w65 8.1.0-release-win32-seh](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-win32/seh/x86_64-8.1.0-release-win32-seh-rt_v6-rev0.7z/download) (which includes `zlib` and `zconf`) leads to compilation issues when building robotgo, as those dependencies are not included in newer MinGW versions.

**Comment by timendez** [comment]
Were you able ever to get past this error? I tried downloading `zlib` manually but still getting this error.

**Comment by ChristoWolf** [comment]
Hi @timendez!

I probably could have worked around it, but lost motivation at some point.

It was missing zlib.h and zconf.h, which I added from the `zlib` download.
That worked out fine, but then it needed libs and `d`-files and I gave up and created this issue.
I could have vendored more things until it would have worked, but I don't want to pollute my environments with all this stuff.

IMO it would be best to update the CGo stuff to require only `std` stuff, or - worst case - to add scripts/[taskfiles](https://github.com/go-task/task)/make-files which do the heavy lifting of the setup.
